Gems/DiffuseProbeGrid/Code/Source/EditorComponents/EditorDiffuseProbeGridComponent.cpp

```diff
@@ -572,8 +573,26 @@ namespace AZ
             m_bakeInProgress = true;
             m_controller.BakeTextures(bakeTexturesCallback);
 
+            QProgressDialog bakeDialog;
+            bakeDialog.setWindowFlags(bakeDialog.windowFlags() & ~Qt::WindowCloseButtonHint);
+            bakeDialog.setLabelText(QObject::tr("Baking Diffuse Probe Grid..."));
+            bakeDialog.setWindowModality(Qt::WindowModal);
+            bakeDialog.setMaximumSize(QSize(256, 96));
+            bakeDialog.setMinimum(0);
+            bakeDialog.setMaximum(0);
+            bakeDialog.setMinimumDuration(0);
+            bakeDialog.setAutoClose(false);
+            bakeDialog.setCancelButton(nullptr);
+            bakeDialog.show();
+
             while (m_bakeInProgress)
             {
+                if (bakeDialog.wasCanceled())
+                {
+                    m_bakeInProgress = false;
+                    break;
+                }
+
                 QApplication::processEvents();
                 AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(100));
```

Comment:
Out of scope for this change, but the baking operation could be moved to use the `AZ::Job` system so that the baking could happen on a separate thread, so you wouldn't need to do this sleep/pump for events on the main thread.